### ISSUE TITLE
[ACT:5c01ea] Some global attributes should be disallowed for the element with 'generic' role

### ISSUE THREAD

**Issue by shunguoy** [IBMa/equal-access#1253]
The engine reports a pass for ACT [Fail Example 3 ](https://www.w3.org/WAI/standards-guidelines/act/rules/5c01ea/proposed/).  This should fail.
 The [global attributes](https://w3c.github.io/aria/#generic) are:
    aria-brailleroledescription
    aria-label
    aria-labelledby
    aria-roledescription

The fix needs to cover the following aspects:
 1.  report an error for an element with an implicit 'generic' role if it has certain global attributes.
 2.  disallow explicit 'generic' role.

**Comment by shunguoy** [comment]
Another example is:
```
<div role="list">
	<div aria-live="polite">
		<div role="listitem">List item 1</div>
		<div role="listitem">List item 2</div>
	</div>
</div>
```
The engine passes this. ACT says it’s a fail because of the div aria-live=“polite”.

**Comment by philljenkins** [comment]
I think this should also be better explained (more straightforward) in the Help for the existing rule 
Rule ID HELP: [aria_attribute_allowed](https://unpkg.com/accessibility-checker-engine@3.1.44/help/en-US/aria_attribute_allowed.html#%7B%22message%22%3A%22The%20ARIA%20attributes%20%5C%22%7B0%7D%5C%22%20are%20not%20valid%20for%20the%20element%20%3C%7B1%7D%3E%20with%20ARIA%20role%20%5C%22%7B2%7D%5C%22%22%2C%22msgArgs%22%3A%5B%22%7B0%7D%22%2C%22%7B1%7D%22%2C%22%7B3%7D%22%2C%22%7B4%7D%22%5D%2C%22value%22%3A%5B%22VIOLATION%22%2C%22FAIL%22%5D%2C%22reasonId%22%3A%22Fail_invalid_role_attr%22%7D)
Reason ID: Fail_invalid_role_attr

**Comment by sct-will** [comment]
The dev work for this may be split into more issues.

**Comment by shunguoy** [comment]
this issue is expanded to include the ARIA "generic" role support, The engine changes include:  (1) aria definition; (2) aria path/accessibility tree; (3) any hierarchy rules (parent to child and child to parent) related to the aria change; and (4) prohibited prop explained in this issue.

**Comment by sct-will** [comment]
Fixing this test case. Need to see what is root cause.

**Comment by ErickRenteria** [comment]
Validated with https://github.com/IBMa/equal-access/pull/1329
